python/desc/sims/GCRCatSimInterface/HostImages.py

```diff
@@ -0,0 +1,76 @@
+import os
+import numpy as np
+import pandas as pd
+from lsst.sims.utils import angularSeparation
+
+__all__ = ['hostImage']
+
+class hostImage(object):
+
+    def __init__(self, ra_center, dec_center, fov):
+
+        self.ra = ra_center
+        self.dec = dec_center
+        self.radius = fov
+
+
+    def format_catalog(self, df_line, fits_file_name, image_dir):
+
+        fits_info = fits_file_name.split('_')
+        lens_id = np.int(fits_info[0])
+        sys_magNorm = np.float(fits_info[1])
+        gal_type = fits_info[2].split('.')[0]
+
+        galaxy_id = np.right_shift(lens_id, 10)
+        galaxy_id *= 10000
+        galaxy_id += 4*df_line['twinkles_system']
+        sys_id = np.left_shift(galaxy_id, 10)
+        if gal_type == 'bulge':
+            sys_id += 97
+        elif gal_type == 'disk':
+            sys_id += 107
+        else:
+            raise ValueError('Invalid Galaxy Component Type in filename')
+
+        cat_str = 'object %i %f %f %f %s %f 0 0 0 0 0 %s 0.01 0\n' % (sys_id,
+                                                                      df_line['raPhoSim_lens'],
+                                                                      df_line['decPhoSim_lens'],
+                                                                      sys_magNorm,
+                                                                      df_line['sedFilepath'],
```

Comment:
Since the code is letting the user specify the image directory (which they may choose to relocate for staging purposes in the simulation pipeline), it seems like this should be `os.path.basename(df_line['sedFilepath'])` so that the image directory can be specified by the user when running phosim.